client/my-sites/plugins/plugin-details.jsx

```diff
@@ -0,0 +1,440 @@
+									isPluginInstalledOnsite
+								),
+							}
+						) }
+					>
+						<div className="plugin-details__header">
+							<div className="plugin-details__price">{ translate( 'Free' ) }</div>
+							<div className="plugin-details__install">
+								<CTA
+									slug={ props.pluginSlug }
+									isPluginInstalledOnsite={ isPluginInstalledOnsite }
+								/>
+							</div>
+							<div className="plugin-details__t-and-c">
+								{ translate(
+									'By installing, you agree to WordPress.com’s Terms of Service and the Third-Party plug-in Terms.'
+								) }
+							</div>
+						</div>
+					</div>
+				</div>
+
+				<div className="plugin-details__sites-list">
+					<SitesList
+						fullPlugin={ fullPlugin }
+						isPluginInstalledOnsite={ isPluginInstalledOnsite }
+						{ ...props }
+					/>
+				</div>
+
+				<div className="plugin-details__layout plugin-details__body">
+					<div className="plugin-details__layout-col plugin-details__layout-col-left">
+						{ fullPlugin.wporg ? (
+							<PluginSections
+								className="plugin-details__plugins-sections"
+								plugin={ fullPlugin }
+								isWpcom={ isWpcom }
+								addBanner
+								removeReadMore
+							/>
+						) : (
+							<PluginSectionsCustom plugin={ fullPlugin } />
+						) }
+					</div>
+					<div className="plugin-details__layout-col plugin-details__layout-col-right">
+						<div className="plugin-details__plugin-details-title">
+							{ translate( 'Plugin details' ) }
+						</div>
+						<div className="plugin-details__plugin-details-content">
+							<div className="plugin-details__downloads">
+								<div className="plugin-details__downloads-text title">
+									{ translate( 'Downloads' ) }
+								</div>
+								<div className="plugin-details__downloads-value value">
+									{ formatNumberCompact( fullPlugin.downloaded, 'en' ) }
+								</div>
+							</div>
+							<div className="plugin-details__tested">
+								<div className="plugin-details__tested-text title">
+									{ translate( 'Tested up to' ) }
+								</div>
+								<div className="plugin-details__tested-value value">{ fullPlugin.version }</div>
+							</div>
+						</div>
+					</div>
+				</div>
+			</div>
+		</MainComponent>
+	);
+}
+
+function shouldDisplayCTA( selectedSite, slug, isPluginInstalledOnsite ) {
+	return (
+		isPluginInstalledOnsite === false &&
+		selectedSite &&
+		userCan( 'manage_options', selectedSite ) &&
+		isCompatiblePlugin( slug )
+	);
+}
+
+function CTA( { slug, isPluginInstalledOnsite } ) {
+	const dispatch = useDispatch();
+	const translate = useTranslate();
+	const selectedSite = useSelector( getSelectedSite );
+	const selectedSiteId = useSelector( getSelectedSiteId );
+
+	if ( ! shouldDisplayCTA( selectedSite, slug, isPluginInstalledOnsite ) ) {
+		return null;
+	}
+
+	const shouldUpgrade = ! (
+		isBusiness( selectedSite.plan ) ||
+		isEnterprise( selectedSite.plan ) ||
+		isEcommerce( selectedSite.plan )
+	);
```

Comment:
Jetpack self hosted sites shouldn't need to upgrade - buy a business plan.

Reply:
This is an example of how we can figure out a Jetpack self hosted site
`https://github.com/Automattic/wp-calypso/blob/038f3877a3e639dceda8ba5e5dcbf6bc18d94632/client/my-sites/plugins/plugins-browser/index.jsx#L567`

Reply:
For what I understood if a site is already jetpack they won't need to upgrade. Updating the code to:

```ts
const isJetpack = useSelector( ( state ) => checkJetpackSite( state, selectedSite.ID ) );
const shouldUpgrade = ! (
    isBusiness( selectedSite.plan ) ||
    isEnterprise( selectedSite.plan ) ||
    isEcommerce( selectedSite.plan ) ||
    isJetpack
);
```

Reply:
Seems good 👌